vertx-template-engines/vertx-web-templ-pug/src/main/java/io/vertx/ext/web/templ/pug/impl/PugTemplateEngineImpl.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.templ.pug.impl;
+
+import de.neuland.pug4j.PugConfiguration;
+import de.neuland.pug4j.template.PugTemplate;
+import de.neuland.pug4j.template.TemplateLoader;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.common.template.CachingTemplateEngine;
+import io.vertx.ext.web.common.template.impl.TemplateHolder;
+import io.vertx.ext.web.templ.pug.PugTemplateEngine;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/**
+ * @author <a href="http://pmlopes@gmail.com">Paulo Lopes</a>
+ * @author <a href="http://tfox.org">Tim Fox</a>
```

Comment:
You can update the author to list your self

Reply:
The file has been copied from vertx-web-templ-jade. To comply with attribution notices requirement of Apache license I've went though the git history of the original file and mention the compete author list of the original file in the new file.